src/_pytest/config/__init__.py

```diff
@@ -999,19 +991,19 @@ class InvocationParams:
             Plugins accessing ``InvocationParams`` must be aware of that.
         """
 
-        args: Tuple[str, ...]
+        args: tuple[str, ...]
         """The command-line arguments as passed to :func:`pytest.main`."""
-        plugins: Optional[Sequence[Union[str, _PluggyPlugin]]]
+        plugins: Sequence[str | _PluggyPlugin] | None
         """Extra plugins, might be `None`."""
-        dir: Path
-        """The directory from which :func:`pytest.main` was invoked."""
+        dir: pathlib.Path
+        """The directory from which :func:`pytest.main` was invoked. :type: pathlib.Path"""
```

Comment:
Is the `:type:` change needed?

Reply:
sphinx is broken about deferred type annotations